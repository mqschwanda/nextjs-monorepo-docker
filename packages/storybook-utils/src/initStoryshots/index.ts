import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
