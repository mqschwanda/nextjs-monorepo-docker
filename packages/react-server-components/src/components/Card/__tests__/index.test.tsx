import { render } from '@testing-library/react';
import path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import {
  Button,
  Card,
  CardTitle,
  CardBody,
  CardActions,
} from '@';

initStoryshots({
  configPath: path.resolve(__dirname, '../../../../../storybook/.storybook'),
  storyKindRegex: /^@mqs\/react-server-components\/components\/Card$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Card />', () => {
      it('renders', () => {
        const testId = Card.name;
        const children = 'children...';

        const { getByTestId, asFragment } = render((
          <Card
            className='test'
          >
            <figure>
              <img
                alt='test'
                src='/assets/images/test.jpg'
              />
            </figure>
            <CardBody>
              <CardTitle>
                { 'Card Title' }
              </CardTitle>
              <p>
                { children }
              </p>
            </CardBody>
            <CardActions>
              <Button>
                { 'Click Me' }
              </Button>
            </CardActions>
          </Card>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
