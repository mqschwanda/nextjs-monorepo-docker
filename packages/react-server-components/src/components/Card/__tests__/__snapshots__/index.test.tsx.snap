// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Card /> renders 1`] = `
<DocumentFragment>
  <div
    class="card shadow-xl card-normal test"
    data-testid="Card"
  >
    <figure>
      <img
        alt="test"
        src="/assets/images/test.jpg"
      />
    </figure>
    <div
      class="card-body"
      data-testid="CardBody"
    >
      <span
        class="card-title"
        data-testid="CardTitle"
      >
        Card Title
      </span>
      <p>
        children...
      </p>
    </div>
    <div
      class="card-actions"
      data-testid="CardActions"
    >
      <button
        class="btn btn-md"
        data-testid="Button"
        type="button"
      >
        Click Me
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card Compact Example 1`] = `
<div
  className="card shadow-xl card-compact"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Action On Top Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-square btn-sm"
        data-testid="Button"
        type="button"
      >
        <svg
          className="fill-none stroke-current stroke-1"
          data-testid="IconClose"
          fill="currentColor"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </button>
    </div>
    <p>
      We are using cookies for no reason.
    </p>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Badge Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
      <div
        className="badge badge-secondary"
      >
        Cute!
      </div>
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Bottom Image Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Centered Content Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body items-center text-center"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Custom Color Example 1`] = `
<div
  className="card shadow-xl card-normal bg-primary text-primary-content"
  data-testid="Card"
>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Glass Example 1`] = `
<div
  className="card shadow-xl card-normal glass"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Image On Side Example 1`] = `
<div
  className="card shadow-xl card-side card-normal"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/250/500"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Image Overlay Example 1`] = `
<div
  className="card shadow-xl image-full card-normal"
  data-testid="Card"
>
  <figure>
    <img
      alt="Kittens"
      src="http://placekitten.com/1000/400"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With Neutral Color Example 1`] = `
<div
  className="card shadow-xl card-normal bg-neutral text-neutral-content"
  data-testid="Card"
>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Card Card With No Image Example 1`] = `
<div
  className="card shadow-xl card-normal"
  data-testid="Card"
>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      Kittens!
    </span>
    <p>
      Kittens learn to walk when they are around three weeks old.
    </p>
    <div
      className="card-actions justify-end"
      data-testid="CardActions"
    >
      <button
        className="btn btn-primary btn-md"
        data-testid="Button"
        type="button"
      >
        See More Kittens
      </button>
    </div>
  </div>
</div>
`;
